crates/viewer/re_data_ui/src/blob.rs

```diff
@@ -0,0 +1,225 @@
+use egui::{NumExt, Vec2};
+use re_log::ResultExt;
+use re_renderer::renderer::ColormappedTexture;
+use re_types::components::{Blob, MediaType};
+use re_ui::{list_item::PropertyContent, UiExt as _};
+use re_viewer_context::{gpu_bridge::image_to_gpu, SystemCommandSender, UiLayout};
+
+use crate::{image::show_image_preview, EntityDataUi};
+
+impl EntityDataUi for Blob {
+    fn entity_data_ui(
+        &self,
+        ctx: &re_viewer_context::ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        ui_layout: UiLayout,
+        entity_path: &re_log_types::EntityPath,
+        row_id: Option<re_chunk_store::RowId>,
+        query: &re_chunk_store::LatestAtQuery,
+        _db: &re_entity_db::EntityDb,
+    ) {
+        let compact_size_string = re_format::format_bytes(self.len() as _);
+
+        // We ignore the logged `MediaType` component, because the user is looking
+        // at the blob specifically, not the entity as a whole!
+        let media_type = MediaType::guess_from_data(self);
```

Comment:
Treating MediaType as an implied data-dependency for Blob and querying it here seems preferable to just guessing. This seems inline with the thing we will need to do with ImageBuffer anyways.

Reply:
Showing the _actual_ mime of the actual blob has the advantage that it is _true_. That is: it can help a user debug if they accidentally log a .png image with a `image/jpeg` media-type.

In general I would prefer it if we did as few of these magical lookups as possible, and instead show the user what they select. If they click on specific component, then only show data about that component!

For `Blob`, the guessed media type should work for all common image formats anyways.

---

For `ImageBuffer` we do _need_ the actually logged `ImageFormat` though, but then we should make that clear that the shown image is inferred from a sibling component, or we show the image when the _entity_ is selected.